### PR TITLE
chore: detekt format rules

### PR DESCRIPTION
**Pull request by yamilmedina**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

This adds the basic detekt rules for formatting, later on we need to apply the rules (since still trying to find a way to formatting take in account baseline.xml file). I'll prepare another PR that should be the starting point for auto-formatting enabled by default.

~~Note: This PR will fail and there should be an upcoming that will run the tool on files as a starting point.~~

### Causes (Optional)

After this, we don't have to look in PR reviews for formatting, extra spaces and so on (less noise to reviews) =)

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [x] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [x] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by vitorhugods** [comment]
@yamilmedina we have a `.editorconfig` file too, which tells our IDEs about the codestyle.
Can we work together to make sure that when we hit `⌥ ⌘ L` or `Ctrl + Alt + L` it follows the codestyle checked by Detekt?

**Comment by yamilmedina** [comment]
> @yamilmedina we have a `.editorconfig` file too, which tells our IDEs about the codestyle in the project. Can we work together to make sure that when we hit `⌥ ⌘ L` or `Ctrl + Alt + L` it follows the codestyle checked by Detekt?

Yeah! I think that's totally possible, I'll submit another commit having this into account, I'll sync with you

**Review by vitorhugods — APPROVED** [review]
In general I'm happy to see this added to the project.

We also should:
- Warn everyone in the team about this
  - Make sure people know about the Detekt plugin and how to set it up
  - Make sure people are aware that they can report and raise changes to the Detekt rules if we have a mismatch between `.editorconfig` and Detekt settings.
- Update Reloaded's `.editorconfig`

File: detekt/detekt.yml [hunk 1]
```diff
@@ -215,114 +215,112 @@ exceptions:
...
+    Indentation:
+        active: false
+        autoCorrect: true
+        indentSize: 4
+    MaximumLineLength:
+        active: true
+        maxLineLength: 140
+    ModifierOrdering:
+        active: false
+        autoCorrect: true
```
[the remark this answers is not in the document]
Comment (yamilmedina) [reply detekt/detekt.yml]: enabled =D

**Pull request merged**